### PR TITLE
refactor(transcript): backfill identity and description only when uniquely provable

### PR DESCRIPTION
**Pull request by LionSR**
## Summary

Stage 4 of #9590: backfill of legacy execution↔stream identity and of execution descriptions must happen only when uniquely provable, never by choosing among candidates. Investigation against current main (post-#9661) found every behavioral rule already holds; this PR is therefore doctrine-made-explicit plus the missing test pins — production changes are comment-only (+18/−3 lines, zero behavior change), with 3 new public-boundary tests and no-backfill assertions added to 2 existing ambiguity tests (+119 test lines).

Findings per sub-item, verified against the diff of #9661 and current main:

1. **Identity backfill hardening — verified already true, now explicit and pinned.** `writeLegacyExecutionStreamId` (the `streamIdSource: LEGACY_RESOLUTION` stamp) has exactly one production caller, inside `resolvePersistedStreamIdForExecution`, guarded by `metaMatched.length === 1 && streamId`: it fires only when exactly one persisted sidecar claims the execution. Every ambiguous resolution (`exactExecutionCandidateStreamIds`, `suffixCandidateStreamIds`, multi-match with a selectable root) and every suffix-only resolution skips backfill. The skip is now stated in a comment at the write site and pinned by tests. The write is best-effort by contract (`persistSupplementaryMetaFieldsBestEffort` never throws).
2. **Description backfill — verified already true, now explicit and pinned.** The only projection is `StreamSnapshotStore.backfillDescriptionsFromExecutionMeta` (private, 2 call sites: `load`/`preload`), and it already runs meta → mirror only, and only when the mirror is absent (`if (!executionId || meta.description) continue`). Nothing anywhere writes `ExecutionMeta.description` from the sidecar mirror (`writeSessionDescription` is the sole meta-description writer, fed only by AI generation in `sessionDescription.ts`). Per the issue's minimality rule, no new machinery was added — the absent-only/authority-direction rule is now the method's documented contract, with tests.
3. **No heuristic root choice — verified already true, no change.** Grep across production code: the only `listPersistedStreams`/`readPersistedStreamAssociation` callers outside the legacy boundary are `SessionStores`' bulk-deletion/orphan-sweep enumeration (Stage 5's charter, not identity resolution); `chatSessionController` reads a single stream's own sidecar (not a scan); `traceAssembler`/`completedRunArchive` consume only resolution-stated `fallbackStreamIds` (empty by proof for registered records). No alternate-root probe remains in any backfill path.
4. **#8226 hazard — verified safe, now pinned.** A display-materialized stream (`StreamLogStore.ensureStream`, ProgressFactApplier paths) has no sidecar and no `runDescriptor.executionId`, so it can never enter the meta-matched set that authorizes backfill; it can only surface via the suffix arm, which never backfills. New test pins exactly this: an `ensureStream`-materialized orphan carrying the `#executionId` suffix resolves for compatibility display but never stamps `LEGACY_RESOLUTION` onto the execution record.

## Issue acceptance gates (#9590 Stage 4)

- **Obligation (7)** — description reconciliation has one winner and is idempotent: two public-boundary tests in `StreamSnapshotStore.vitest.ts`. An absent mirror is filled from meta and a second `load()` runs zero projections (spy-proven no-op); when both sides exist and disagree, neither side is written — meta keeps its authoritative value, the mirror stays display data — and a mirror-only record never backfills meta.
- **Skip-backfill-on-ambiguity** — assertions added to both ambiguity shapes at the resolver boundary (two unproven sidecar roots; two suffix-only candidates): no `streamId` is ever stamped on the execution record.
- **#8226 hazard check** — result: candidate sets already exclude display-materialized streams from provable (meta-matched) backfill by construction; the new orphan-stream test pins that the suffix arm's compatibility resolution never persists identity.

## Single source of truth

`ExecutionMeta.description` is the description authority (A4); the snapshot sidecar `description` is a display-only mirror. Projection direction is meta → mirror, absent-only; disagreement is not reconciled by writing either side. Identity authority is unchanged from Stages 2–3: `ExecutionMeta.streamId` (registration provenance) forward, sidecar `RunDescriptor.executionId` reverse; `LEGACY_RESOLUTION` stamps are minted only from a uniquely-provable sidecar claim.

## Duplication and abstraction budget

Zero new abstractions, helpers, wrappers, or facades. No new machinery for meta-from-mirror backfill (the codebase never had it and the issue forbids inventing it). The production diff is 2 comment blocks at the 2 backfill sites.

## Net elements (R6)

- Public methods/exports added: 0. Removed: 0.
- Production functions added: 0. Production lines: +18/−3, all documentation at the two backfill sites.
- Tests: +3 new `it` blocks, +2 assertions in existing tests, +4 seeded stamp-able meta records so the no-backfill pins can fail (Bugbot finding, fixed in 9afc375308) — +119 lines across 2 suites.

## Consumer counts (R8)

- `writeLegacyExecutionStreamId`: 1 production caller (the boundary's unique-proof branch) — unchanged.
- `backfillDescriptionsFromExecutionMeta`: private, 2 internal call sites (`load`, `preload`) — unchanged.
- `writeSessionDescription`: 1 production caller (`sessionDescription.ts`) — unchanged.

## Host impact

None. All touched files are host-agnostic `src/transcript/` plus centralized tests; extension, desktop, and CLI behavior is byte-identical (production changes are comments).

## Scheduled refactoring

Stage 6 (#9590) stops the duplicate meta → mirror description projection for current records (the mirror then converges to a pure display cache); this PR's contract comment names that handoff at the projection site. Stage 5 gives `SessionStores`' deletion/sweep enumeration its explicit statement; Stage 7 (#9627) deletes the boundary's scan/suffix arms at the dated retirement (≥ 2026-11-01), gated on `LegacyIdentity` warn-log evidence.

## Validation

- `npm run typecheck` — clean.
- `env -u FORCE_COLOR npx vitest run` on both touched suites — 92/92 passed.
- `npm run check:dead-code-ratchet` — 18 findings vs 18 baselined, no widening.
- `npm run format` (prettier) on all touched files.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/test-kernel/transcript/legacyExecutionIdentity.vitest.ts [hunk 1]
```diff
@@ -347,4 +356,26 @@ describe('resolvePersistedStreamIdForExecution', () => {
...
+    await expect(
+      resolvePersistedStreamIdForExecution(executionId, {
+        snapshotStore: new StreamSnapshotStore(),
+        streamLogStore: logs,
+      }),
+    ).resolves.toEqual({ streamId: orphanStream, fallbackStreamIds: [] });
+    expect(
+      (await getExecutionStore(executionId).readMeta())?.streamId,
+    ).toBeUndefined();
+  });
```
[the remark this answers is not in the document]
Comment (LionSR) [reply src/test-kernel/transcript/legacyExecutionIdentity.vitest.ts:400]: Real finding — fixed in 9afc375308. The backfill write (persistSupplementaryMetaFieldsBestEffort) indeed no-ops on absent metadata, so the no-stamp assertions could not fail. Each of the four pins (two ambiguity shapes, the sidecar/log candidate test, and the orphan-stream test) now seeds a minimal stamp-able ExecutionMeta record before resolving, so a wrongful LEGACY_RESOLUTION stamp would persist and trip the assertion.
